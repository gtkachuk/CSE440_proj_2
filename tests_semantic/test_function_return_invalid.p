program testFunctionReturnInvalid;

class testFunctionReturnInvalid

BEGIN
   VAR someVar : integer;
FUNCTION testFunctionReturnInvalid: integer;
BEGIN
   testFunctionReturnInvalid := 3;
   someVar := testFunctionReturnInvalid
END

END
.
