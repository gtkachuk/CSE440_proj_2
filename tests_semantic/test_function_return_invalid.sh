#!/bin/sh

. ../tests_lib.inc

test_diff_sem test_function_return_invalid.p
