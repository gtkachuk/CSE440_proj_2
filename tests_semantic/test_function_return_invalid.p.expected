ERROR:11:Variable name "testFunctionReturnInvalid" is reserved for use as a function return value
Errors detected. Exiting.
